File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/CoverageDefinition.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// CoverageDefinition Declaration
//==============================================================
type CoverageDefinition struct {
    gorm.Model
     Name                                    string
    DefaultLimit                                                            string
    DefaultDeductible                                                            string
    AsMandatory                                    bool
    ProductId         *uint
    Product           *InsuranceProduct `gorm:"foreignKey:ProductId"`
    CoverageType                      CoverageType

// parent associations as their child

}

File: evidence/generated-systems/django/insurance-on-django/insuranceOnDjango/models/CoverageDefinition.py
from django.db import models
from insuranceOnDjango.models.CoverageType import CoverageType

#======================================================================
# 
# Encapsulates data for model CoverageDefinition
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class CoverageDefinition Declaration
#======================================================================
class CoverageDefinition (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	defaultLimit = Money
	defaultDeductible = Money
	asMandatory = models.BooleanField(null=True)
	product = models.ForeignKey('InsuranceProduct', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	coverageType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CoverageType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.defaultLimit
		str = str + self.defaultDeductible
		str = str + self.asMandatory
		str = str + self.coverageType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "CoverageDefinition";
    
	def objectType(self):
		return "CoverageDefinition";
